NumLib/ODESolver/MatrixTranslator.h

```diff
@@ -12,374 +12,249 @@
 
 #include <memory>
 
-#include "Types.h"
 #include "TimeDiscretization.h"
+#include "Types.h"
 
 #include "MathLib/LinAlg/BLAS.h"
 
 namespace NumLib
 {
-
 //! \addtogroup ODESolver
 //! @{
 
-
-/*! Translates matrices and vectors assembled by a provided ODE (or other equation)
- *  to the stiffness matrix and right-hand side vector of a linear equation system
- *  that can be solved by a linear equation system solver.
+/*! Translates matrices and vectors assembled by a provided ODE (or other
+ * equation) to the stiffness matrix and right-hand side vector of a linear
+ * equation system that can be solved by a linear equation system solver.
  *
- * \tparam Matrix the type of matrices occuring in the linearization of the equation.
- * \tparam Vector the type of the solution vector of the equation
  * \tparam ODETag a tag indicating the type of equation.
  */
-template<typename Matrix, typename Vector, ODESystemTag ODETag>
+template <ODESystemTag ODETag>
 class MatrixTranslator;
 
-
-/*! Translates matrices assembled by a provided first order implicit quasi-linear ODE
- *  to some other matrices suitable to be passed on to nonlinear solvers.
- *
- * \tparam Matrix the type of matrices occuring in the linearization of the equation.
- * \tparam Vector the type of the solution vector of the equation
+/*! Translates matrices assembled by a provided first order implicit
+ * quasi-linear ODE to some other matrices suitable to be passed on to nonlinear solvers.
  *
  * \see ODESystemTag::FirstOrderImplicitQuasilinear
  */
-template<typename Matrix, typename Vector>
-class MatrixTranslator<Matrix, Vector, ODESystemTag::FirstOrderImplicitQuasilinear>
+template <>
+class MatrixTranslator<ODESystemTag::FirstOrderImplicitQuasilinear>
```

Comment:
No, the specializations are entirely different types. So the declarations of the members must stay here.